app/experimenter/nimbus-ui/src/components/PageHome/SearchBar/index.tsx

```diff
@@ -66,7 +91,7 @@ const SearchBar: React.FunctionComponent<SearchBarProps> = ({
       setClearIcon(true);
 
       const newTimer = setTimeout(() => {
-        const results = fuse.search(searchTerms);
+        const results = fuse.search(event.target.value as string);
```

Comment:
Hi @yashikakhurana , thank you for the suggestion. It was actually what I did before. But it takes some time for the Fuse function to search with the term and setting the searchTerm state before the fuse results returned would not trigger re-render. It will search the term on empty data. That is why I went with the current solution which seems a bit redundant.

Reply:
the only difference is to check the history of the search exists in URL, if yes we will initialize it, if it's not present, it will use as it was running before, even on the first rendering history doesn't have a search term, but it will benefit second time, let me know if you have any more questions :)

Reply:
Yes, but we also want the form to change accordingly with the search term, meaning we not only want the search bar to display the term but also to return to the page where search results are displayed with it. Using the setSearchTerm will not trigger the form re-rendering with the new search term. 
The form re-renders on OnChange instead of the change of the value. Therefore, simply update the search term state would not trigger the form to change. That is why I tried to dispatch an event to trigger handleChange manually. I hope I stated clear and understood correctly of what we want achieve with this feature.

Reply:
I see what you mean, fair enough but I will still suggest instead of this fuse.search(event.target.value as string) use states here

Reply:
Thank you for the understanding. Right, I tried call fuse.search again as well. Yet, The problem I faced was fuse itself has not yet been initialized. I printed fuse and the _docs array was empty. I thought the timer logic of handleChange makes a lot of sense so I did not modify it too much or repeat it but tried to trigger it. Yet, simply triggering it with input event still resulted in the same problem unless I setTimeout. Those were the steps I went through. But you are right, I will look deeper and I also would like to do this more elegantly. I will also resolve the conflicts.

Reply:
but I would say you are on right track 👏

Reply:
https://github.com/mozilla/experimenter/pull/8418#discussion_r1130280797 Hi Yashika, regarding this line, the states will not be updated as the target.value at here yet. Because the set state in React will not synchronize instantly. For example, if the input is "Nam" the state here will still be "Na" as that is the last state of searchTerms. Is this what we intend to do here? I just want to double check to be sure of it.

Reply:
so we want to have searchterm but we also want to update results based on that, the other approach you can also do that having a state of search term outside the search bar component and then pass it as a prop in that way results will stored

Reply:
Hi I didn't see why making it a prop can solve any thing. So what we have is 1) line97 - the handleChange function always need to use the event.target.value for update because setState does not happen immediately and updating with the current value as well as setState for the state is the way FormControl works. So this has to be changed and we just cannot use the state here to search here. 2) new Fuse() is not actually synchronized but a-synchronized. So the useEffect and setTimeout way of handling. I couldn't see any other way to solve this unless you meant making the Fuse result as a prop to pass in this function. Could you please be more specific about which part you want me to change?